it/uk/gov/hmrc/fileupload/DownloadEnvelopeIntegrationSpec.scala

```diff
@@ -59,6 +60,48 @@ class DownloadEnvelopeIntegrationSpec extends IntegrationSpec with EnvelopeActio
         response.body.contains("sampleFileContent") shouldBe true
       }
     }
+  }
+
+
+  feature("Download Envelope MongoDB") {
+
+    scenario("A client can download an envelope including its file") {
+      Given("I have an envelope with files")
+      val envelopeId = createEnvelope()
+      val data = "my file content"
+      val fileId = FileId(s"fileId-${nextId()}")
+      val fileRefId = FileRefId()
```

Comment:
Not related to this ticket but still valuable:
TODO: FileRedId should not have a default value anymore